crates/virtio-queue/src/queue.rs

```diff
@@ -668,14 +665,15 @@ mod tests {
         vq.avail().ring().ref_at(2).store(u16::to_le(5));
         // Let the device know it can consume chains with the index < 2.
         vq.avail().idx().store(u16::to_le(3));
+
         // No descriptor chains are consumed at this point.
-        assert_eq!(q.next_avail(), 0);
-        assert_eq!(q.next_used(), 0);
+        assert_eq!(state.next_avail(), 0);
```

Comment:
The problem with this refactoring is that you are no longer calling these methods from the `Queue`/`QueueGuard` implementation, but always from the `QueueState` one. Because of this, having two separate tests doesn't actually make much sense anymore. I'll think about what we can do regarding this.

Reply:
I didn't find any solution to this without some serious refactoring. One way of achieving this is to implement the same trait on both QueueGuard and Queue, and the test case receive as a parameter a parametrized Q where Q implements the common trait. This is extending far beyond the scope of the opened issue. The code in Queue and QueueGuard is duplicated as well, but I think that needs to be handled as a separate issue

Reply:
> I didn't find any solution to this without some serious refactoring. One way of achieving this is to implement the same trait on both QueueGuard and Queue, and the test case receive as a parameter a parametrized Q where Q implements the common trait. This is extending far beyond the scope of the opened issue. The code in Queue and QueueGuard is duplicated as well, but I think that needs to be handled as a separate issue

I agree, so should we drop this patch for now?